packages/wallet-sdk/src/core/analytics/logEvent.ts

```diff
@@ -0,0 +1,100 @@
+import { store } from ':store/store.js';
+import { VERSION } from '../../sdk-info.js';
+
+enum ComponentType {
+  unknown = 'unknown',
+  banner = 'banner',
+  button = 'button',
+  card = 'card',
+  chart = 'chart',
+  content_script = 'content_script',
+  dropdown = 'dropdown',
+  link = 'link',
+  page = 'page',
+  modal = 'modal',
+  table = 'table',
+  search_bar = 'search_bar',
+  service_worker = 'service_worker',
+  text = 'text',
+  text_input = 'text_input',
+  tray = 'tray',
+  checkbox = 'checkbox',
+  icon = 'icon',
+}
+
+enum ActionType {
+  unknown = 'unknown',
+  blur = 'blur',
+  click = 'click',
+  change = 'change',
+  dismiss = 'dismiss',
+  focus = 'focus',
+  hover = 'hover',
+  select = 'select',
+  measurement = 'measurement',
+  move = 'move',
+  process = 'process',
+  render = 'render',
+  scroll = 'scroll',
+  view = 'view',
+  search = 'search',
+  keyPress = 'keyPress',
+  error = 'error',
+}
+
+enum AnalyticsEventImportance {
+  low = 'low',
+  high = 'high',
+}
+
+type CCAEventData = {
+  // Standard Attributes
+  action: ActionType;
+  componentType: ComponentType;
+  // Metadata
+  sdkVersion: string;
+  appName: string;
+  appOrigin: string;
+  // Custom Attributes
+  method?: string; // RPC method
+};
```

Comment:
what about info about the connected wallet. Like wallet name/id ? (Coinbase Smart Wallet / COINBASE_SMART_WALLET)